plugins/modules/vios_update_upgrade.py

```diff
@@ -0,0 +1,500 @@
+                    - When the VIOS update image is on a remote server or a USB data storage device and the --save option is specified,
+                      this option is required to specify the name used to save the image to the HMC hard disk.
+                type: str
+            files:
+                description:
+                    - The list of the files that is required for update/upgrade
+                    - This option is required and only valid when the VIOS installation image is on a remote server.
+                type: list
+                elements: str
+            host_name:
+                description:
+                    - The host name or IP address of the remote server.
+                type: str
+            user_id:
+                description:
+                    - The user ID to use to log in to the remote SFTP server.
+                type: str
+            password:
+                description:
+                    - The password to use to log in to the remote SFTP server.
+                    - password, ssh_key_file are mutually exclusive.
+                type: str
+            ssh_key_file:
+                description:
+                    - The name of the file that contains the SSH private key.
+                    - password, ssh_key_file are mutually exclusive.
+                type: str
+            mount_loc:
+                description:
+                    - The mount location defined on the NFS server that contains the VIOS update image.
+                type: str
+            option:
+                description:
+                    - Options to be passed to the mount command used to mount the NFS file system that contains the VIOS update image.
+                    - The HMC supports NFS versions 3 and 4.
+                    - Default version is 3.
+                type: str
+                choices: ['3', '4']
+            directory:
+                description:
+                    - The name of the directory on the remote server that contains the VIOS update image.
+                    - If this option is not specified when the VIOS update image is on a SFTP server,
+                      the image will be obtained from the user home directory.
+                    - If this option is not specified when the VIOS update image is in an NFS file system,
+                      the image will be obtained from the mount-location on the NFS server.
+                type: str
+            disks:
+                description:
+                    - The name of one or more free VIOS disks to be used for the upgrade.
+                    - The total of the specified disk sizes must be a minimum of 30 GB.
+                type: list
+                elements: str
+            restart:
+                description:
+                    - Restarts the VIOS after installing the update if the update requires a restart.
```

Comment:
Does this work only for update?, we have to modify if it works for upgrade also

Reply:
done, it works only for update